terraform/provisioning/common.bash.tmpl

```diff
@@ -5,18 +5,18 @@ set -o nounset
 set -o pipefail
 
 common::exit_error() {
-  local -r message="${1:?}"
-  local -r code="${2:-1}"
-  echo "${message}" >&2
-  exit "${code}"
+  local -r message="$${1:\?}"
+  local -r code="$${2:-1}"
+  echo "$${message}" >&2
+  exit "$${code}"
```

Comment:
Instead of templating the whole file, and making it difficult to run manually the script on nodes for manual tests, I would pass a new variable `$contained_af_image` in the same way it has been done for the `$domain` variable. See:

terraform/provision.tf
```
  provisioner "remote-exec" {
    inline = "domain='${local.domain}' bash ~/provisioning/${var.distro}.bash"
  }
```